### ISSUE TITLE
Touchscreen calibration

### ISSUE THREAD

**Issue by jordi-s** [giovannicaligaris/iwork10-pro-linux#1]
Greetings,

I had some success calibrating the screen with the following procedure:

- Run `xinput_calibration -v`
- Add the section that xinput_calibration suggests to `/usr/share/X11/xorg.conf.d/99-calibration.conf`. Also take note of the coordinates of points 0 (x0, y0) and point 3 (x3, y3) as given by xinput_calibration.
- Now leave as the only option in that section `Option "CalibrationMatrix" "a 0 c 0 e f 0 0 1"` where numbers a, c, e and f are calculated as follows
`width=1920`
`height=1200`
`a = (width * 6 / 8) / (x3 - x0)`
`c = ((width / 8) - (a * x0)) / width`
`e = (height * 6 / 8) / (y3 - y0)`
`f = ((height / 8) - (e * y0)) / height`
For example, in my case the result was
`Option "CalibrationMatrix" "2.0571 0 0.005 0 -2.7108 0.99247 0 0 1" `

That is enough to make the touchscreen work more or less, depending on the accuracy of the input when running `xinput_calibration`

I hope that it helps!

**Comment by junocomp** [comment]
Did you try this on Ubuntu 18.04?

**Comment by jordi-s** [comment]
Yes (Kubuntu 18.04 to be precise, installed on the i15-TD model), but I suppose that it can be applied to any recent x.org install.

**Comment by jordi-s** [comment]
I attach my configuration file just in case it is useful:
[99-calibration.conf.zip](https://github.com/giovannicaligaris/iwork10-pro-linux/files/3991796/99-calibration.conf.zip)

**Comment by junocomp** [comment]
Thank you. Which touch driver did you use? Can you share that as well please.

**Comment by jordi-s** [comment]
I am not sure if I obtained the driver in the same way as described in this repository.

First, I obtained the `SileadTouch.sys` windows driver. Then, with the following [perl script](https://github.com/onitake/gsl-firmware/blob/master/tools/scanwindrv) I extracted a file that was automatically named `firmware_00.fw`. I renamed it to `mssl1680.fw` and copied it to ` /lib/firmware/silead/mssl1680.fw`. After reboot, I had the touchscreen recognized, but without calibration.
